core/trino-main/src/test/java/io/trino/execution/executor/TestTaskExecutor.java

```diff
@@ -498,6 +498,32 @@ public void testMinDriversPerTaskWhenTargetConcurrencyIncreases()
         }
     }
 
+    @Test
+    public void testWaitingSplitsSizeDistribution()
+    {
+        MultilevelSplitQueue splitQueue = new MultilevelSplitQueue(2);
+        TestingTicker ticker = new TestingTicker();
+        TaskExecutor taskExecutor = new TaskExecutor(4, 1, 2, 2, splitQueue, ticker);
+
+        TaskHandle testTaskHandle = taskExecutor.addTask(
+                new TaskId(new StageId(new QueryId("test"), 0), 0, 0),
+                () -> 0,
+                1,
+                new Duration(1, MILLISECONDS),
+                OptionalInt.of(2));
+        TaskHandle testTaskHandle2 = taskExecutor.addTask(
+                new TaskId(new StageId(new QueryId("test"), 1), 0, 0),
+                () -> 0,
+                1,
+                new Duration(1, MILLISECONDS),
+                OptionalInt.of(2));
+        TestingJob job = new TestingJob(ticker, new Phaser(), new Phaser(), new Phaser(), 1, 0);
+        taskExecutor.enqueueSplits(testTaskHandle, false, ImmutableList.of(job));
```

Comment:
how about adding many jobs at once e.g. `ImmutableList.of(job, job, job)`?

Reply:
You need a different `taskHandle`. I was trying to do it.